app/src/main/java/fr/free/nrw/commons/explore/map/ExploreMapFragment.java

```diff
@@ -156,43 +157,34 @@ public class ExploreMapFragment extends CommonsDaggerSupportFragment
     @BindView(R.id.category)
     TextView distance;
 
-    private ActivityResultLauncher<String[]> activityResultLauncher = registerForActivityResult(
-        new ActivityResultContracts.RequestMultiplePermissions(),
-        new ActivityResultCallback<Map<String, Boolean>>() {
-            @Override
-            public void onActivityResult(Map<String, Boolean> result) {
-                boolean areAllGranted = true;
-                for (final boolean b : result.values()) {
-                    areAllGranted = areAllGranted && b;
-                }
-
-                if (areAllGranted) {
-                    locationPermissionGranted();
+    private ActivityResultLauncher<String> activityResultLauncher = registerForActivityResult(
+        new ActivityResultContracts.RequestPermission(), isGranted -> {
+            if (isGranted) {
+                locationPermissionGranted();
+            } else {
+                if (shouldShowRequestPermissionRationale(permission.ACCESS_FINE_LOCATION)) {
+                    DialogUtil.showAlertDialog(getActivity(),
+                        getActivity().getString(R.string.location_permission_title),
+                        getActivity().getString(R.string.location_permission_rationale_nearby),
+                        getActivity().getString(android.R.string.ok),
+                        getActivity().getString(android.R.string.cancel),
+                        () -> {
+                            askForLocationPermission();
+                        },
+                        null,
+                        null,
+                        false);
                 } else {
-                    if (shouldShowRequestPermissionRationale(permission.ACCESS_FINE_LOCATION)) {
-                        DialogUtil.showAlertDialog(getActivity(),
-                            getActivity().getString(R.string.location_permission_title),
-                            getActivity().getString(R.string.location_permission_rationale_nearby),
-                            getActivity().getString(android.R.string.ok),
-                            getActivity().getString(android.R.string.cancel),
-                            () -> {
-                                if (!(locationManager.isNetworkProviderEnabled()
-                                    || locationManager.isGPSProviderEnabled())) {
-                                    showLocationOffDialog();
-                                }
-                            },
-                            () -> isPermissionDenied = true,
-                            null,
-                            false);
-                    } else {
-                        isPermissionDenied = true;
+                    if (isPermissionDenied) {
+                        locationPermissionsHelper.showAppSettingsDialog(getActivity(),
+                            R.string.nearby_needs_location);
```

Comment:
Reusing the nearby string here is resulting in the incorrect "Nearby needs location enabled to work properly" message being shown. In reality, we really don't need to show a message here at all, right? The explore map works totally fine without location access. I suppose that's why we never showed a message here before.

Reply:
I agree with you, reusing nearby string seems to be a mistake, I'll change that. But this code here is not to show a toast message to the user. It's actually to show a dialog to prompt the user to open the app's settings and give the location permission. This runs only when the user has denied permission twice but has still clicked the target button. Toast will be shown when presses cancel button on this dialog

Reply:
Got it. :+1:

BTW, I wonder if we need to also switch the string we use in the previous `if`. Specifically [this line](https://github.com/commons-app/apps-android-commons/pull/5494/files#diff-01b9e13613728c294f3d18da7008fd3b0065a501b7a222b0cc5e620a2aeae32dR168). It reads as follows:

> Permission required to display a list of nearby places

In reality, it should kind of read as follows isn't it?

> Permission required to display a list of nearby images

Reply:
Yeah, correct, I'll change that :)